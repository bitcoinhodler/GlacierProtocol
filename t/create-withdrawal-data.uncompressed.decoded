{
  "txid": "88cdbe61909d148b030c045428e22e88ba7623bf0d1d715f4b81deab0f47f3ff",
  "hash": "88cdbe61909d148b030c045428e22e88ba7623bf0d1d715f4b81deab0f47f3ff",
  "version": 2,
  "size": 534,
  "vsize": 534,
  "weight": 2136,
  "locktime": 0,
  "vin": [
    {
      "txid": "6bcc025149e99ca932d3b7d6244a1a57735fc06c889e3578ae695a1bbb88bcd8",
      "vout": 1,
      "scriptSig": {
        "asm": "0 304402206a81afff6ae03dd6dfb7b19259f9ae8767b314ea5dfeeec38a0c20d5d59907b902200bf35f6dfd2932cfb825461be90a974f01106b7e28847e00d028b82f16b24653[ALL] 304402202a06727bfb356a888e19a15f71a176b9f265106dcf2f9fc121abbca9db876c2902200dc3de4eb041c730445da03f94afd46099bfe9f1ee26f78384fc8d669f18e2e6[ALL] 52410477f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece1781135cf0f28f643d3c60a75c29db0819d5442bae83d8f74e7c44ced3255665998e52410403f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac931c108c7c3189b7fc7ea7097b02217d819e866a9841291c522f69173a12524a41046895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c674df743bbb58cea21170831334e4d6d965ed34ce23249557091f440305ab9c0041049f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff6f21fcb5a958839fee0022e8ea5ddc6d71fc624a84bdc42835193abfee125b1c354ae",
        "hex": "0047304402206a81afff6ae03dd6dfb7b19259f9ae8767b314ea5dfeeec38a0c20d5d59907b902200bf35f6dfd2932cfb825461be90a974f01106b7e28847e00d028b82f16b246530147304402202a06727bfb356a888e19a15f71a176b9f265106dcf2f9fc121abbca9db876c2902200dc3de4eb041c730445da03f94afd46099bfe9f1ee26f78384fc8d669f18e2e6014d0b0152410477f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece1781135cf0f28f643d3c60a75c29db0819d5442bae83d8f74e7c44ced3255665998e52410403f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac931c108c7c3189b7fc7ea7097b02217d819e866a9841291c522f69173a12524a41046895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c674df743bbb58cea21170831334e4d6d965ed34ce23249557091f440305ab9c0041049f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff6f21fcb5a958839fee0022e8ea5ddc6d71fc624a84bdc42835193abfee125b1c354ae"
      },
      "sequence": 4294967295
    }
  ],
  "vout": [
    {
      "value": 0.00040660,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_HASH160 762fea3246dc4ce19cfd3f1aa9685e8838605057 OP_EQUAL",
        "hex": "a914762fea3246dc4ce19cfd3f1aa9685e883860505787",
        "reqSigs": 1,
        "type": "scripthash",
        "addresses": [
          "2N42986nuqGfmbRUuojYCboE7EE9VLmYai2"
        ]
      }
    },
    {
      "value": 0.00004000,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 fb5b6788a336c0cf49b90e51ad426f2e89bf43fb OP_EQUALVERIFY OP_CHECKSIG",
        "hex": "76a914fb5b6788a336c0cf49b90e51ad426f2e89bf43fb88ac",
        "reqSigs": 1,
        "type": "pubkeyhash",
        "addresses": [
          "n4S1Q6S9b5dq72ttpnFZXwaPCs2Z8ojwPR"
        ]
      }
    }
  ]
}
