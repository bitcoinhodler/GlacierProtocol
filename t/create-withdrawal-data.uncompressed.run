#!/bin/bash
set -e

# Also tests excessive fee correction

$GLACIERSCRIPT --regtest=$1 create-withdrawal-data << INPUT 2>&1
y
y
y
y
y
y
2N42986nuqGfmbRUuojYCboE7EE9VLmYai2
52410477f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece1781135cf0f28f643d3c60a75c29db0819d5442bae83d8f74e7c44ced3255665998e52410403f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac931c108c7c3189b7fc7ea7097b02217d819e866a9841291c522f69173a12524a41046895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c674df743bbb58cea21170831334e4d6d965ed34ce23249557091f440305ab9c0041049f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff6f21fcb5a958839fee0022e8ea5ddc6d71fc624a84bdc42835193abfee125b1c354ae
n4S1Q6S9b5dq72ttpnFZXwaPCs2Z8ojwPR
1
0200000001b0dbf28adfd58c7814f1d91b422f8e8659d3c05dfa9220b17a9d982d79cfb94e010000006a47304402200d0bbd2abd6f91fdd1e7232746cb54742619751f75dc94818b3ac6ba5e82a0a602201e3dd4770b6670884104772f87ede0369caadc08cbc6dd8692ed4029947cd992012102c745d1c5e15240939f10b29836741c9b809f6874302f633c21bd95671e7d7b97ffffffff02300cf505000000001976a914dacd1afaae47b1c4b9adbbb261df55ebf898b38088ac50c300000000000017a914762fea3246dc4ce19cfd3f1aa9685e88386050578700000000
2
92jodMpEwGtS11sMjxfq6acnc1Yi6VX2QFZWr4YQVjXnQGdKH1g
922F7weRdtBgCnqcnuMo8hba8wj75fZc2nx4oTnLHevh9GaCVXA
10000
10
y
0.00004000
y
INPUT
