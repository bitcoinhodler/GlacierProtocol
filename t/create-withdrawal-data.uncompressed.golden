Are you running this on a computer WITHOUT a network connection of any kind? (y/n)?Have the wireless cards in this computer been physically removed? (y/n)?Are you running on battery power? (y/n)?Are you running on an operating system booted from a USB drive? (y/n)?Is your screen hidden from view of windows, cameras, and other people? (y/n)?Are smartphones and all other nearby devices turned off and in a Faraday bag? (y/n)?
You will need to enter several pieces of information to create a withdrawal transaction.


*** PLEASE BE SURE TO ENTER THE CORRECT DESTINATION ADDRESS ***


Source cold storage address: 
Redemption script for source cold storage address: 
Destination address: 
How many unspent transactions will you be using for this withdrawal? 
Please paste raw transaction #1 (hexadecimal format) with unspent outputs at the source address
OR
input a filename located in the current directory which contains the raw transaction data
(If the transaction data is over ~4000 characters long, you _must_ use a file.):

Transaction data found for source address.
TOTAL unspent amount for this raw transaction: 0.00050000 BTC

How many private keys will you be signing this transaction with (at least 2 required to complete transaction)?
#: Key #1: Key #2: 
Enter fee rate.
Satoshis per vbyte: Calculated fee (0.05340000) is too high. Must be under 0.005

Enter fee rate.
Satoshis per vbyte: 
Based on the provided rate, the fee will be 0.00005340 bitcoin.
Confirm? (y/n): 
Please enter the decimal amount (in bitcoin) to withdraw to the destination address.

Example: For 2.3 bitcoins, enter "2.3".

After a fee of 0.00005340, you have 0.00044660 bitcoins available to withdraw.

*** Technical note for experienced Bitcoin users:  If the withdrawal amount & fee are cumulatively less than the total amount of the unspent transactions, the remainder will be sent back to the same cold storage address as change. ***

Amount to send to n4S1Q6S9b5dq72ttpnFZXwaPCs2Z8ojwPR (leave blank to withdraw all funds stored in these unspent transactions): 0.00040660 being returned to cold storage address address 2N42986nuqGfmbRUuojYCboE7EE9VLmYai2.

Is this data correct?
*** WARNING: Incorrect data may lead to loss of funds ***

0.00050000 BTC in unspent inputs from cold storage address 2N42986nuqGfmbRUuojYCboE7EE9VLmYai2
0.00040660 BTC going back to cold storage address 2N42986nuqGfmbRUuojYCboE7EE9VLmYai2
0.00004000 BTC going to destination address n4S1Q6S9b5dq72ttpnFZXwaPCs2Z8ojwPR
Fee amount: 0.00005340

Signing with private keys: 
92jodMpEwGtS11sMjxfq6acnc1Yi6VX2QFZWr4YQVjXnQGdKH1g
922F7weRdtBgCnqcnuMo8hba8wj75fZc2nx4oTnLHevh9GaCVXA


Confirm? (y/n): 
Calculating transaction...

Final fee rate: 10.00 satoshis per vbyte

Raw signed transaction (hex):
02000000019f5ba6bd00f6307586c12bab98517883b8b5927acb5c3da43e02224a06f9876501000000fd9f010047304402205cad7e9a0616616f3bfb23d76dc8768af3aa3f9c25c7b9f085bfbc35011f81da02205b48db8dfeab3049ce00b8587ce4789027ff2f1ebc1c59c892347794440b97630147304402205f321d860ee7a6e7aab91ebe01fd81024291c8e1d3ce0ff3265f30807d424a930220497091ed26bb22137d5c95215bc5946ef5d000c2c442ac94c3c930f2334c119a014d0b0152410477f15f22aeffaf3f3bc48a280a767f7c6af21276783c09ff3bcaabbece1781135cf0f28f643d3c60a75c29db0819d5442bae83d8f74e7c44ced3255665998e52410403f642223bc751e125ab604ed20c544e933eb82a19201f90bda6236a9ddbedac931c108c7c3189b7fc7ea7097b02217d819e866a9841291c522f69173a12524a41046895f3ed01c09025ba9cfab7b67d699e3ebaceada0beff70bc72c7f10df2d1c674df743bbb58cea21170831334e4d6d965ed34ce23249557091f440305ab9c0041049f3cb152130c241e01b9220a147f69f7111b830def35b510973ab47bb9af7ff6f21fcb5a958839fee0022e8ea5ddc6d71fc624a84bdc42835193abfee125b1c354aeffffffff02d49e00000000000017a914762fea3246dc4ce19cfd3f1aa9685e883860505787a00f0000000000001976a914fb5b6788a336c0cf49b90e51ad426f2e89bf43fb88ac00000000

Transaction fingerprint (md5):
dd017f0a5eeac4f5b061fddc4431cbf7
QR code for transaction written to transaction.png
